src/js/containers/Submission.jsx

```diff
@@ -16,11 +16,46 @@ import Signature from './Signature.jsx'
 import Summary from './Summary.jsx'
 import RefileButton from '../containers/RefileButton.jsx'
 import ParseErrors from './ParseErrors.jsx'
+import LoadingIcon from '../components/LoadingIcon.jsx'
 
 const EditsNav = submissionProgressHOC(EditsNavComponent)
 const NavButton = submissionProgressHOC(NavButtonComponent)
 const RefileWarning = submissionProgressHOC(RefileWarningComponent)
 
+const renderByCode = (code, page, message) => {
+  const toRender = []
+  if(code === -1) {
+    toRender.push(<p>{message}</p>)
+  }else{
+    if(page === 'upload'){
+      toRender.push(<UploadForm code={code}/>)
+      if(code === 5) {
+        toRender.push(<RefileWarning/>)
+        toRender.push(<ParseErrors/>)
+      }
+    }else if(['syntacticalvalidity','quality','macro'].indexOf(page) !== -1){
+      if(code > 6){
+        if(code === 8) toRender.push(<RefileWarning/>)
+        toRender.push(<Edits/>)
+      }
+    }else if(page === 'summary'){
+      if(code > 7){
+        toRender.push(<IRSReport/>)
+        toRender.push(<Summary/>)
+        toRender.push(<Signature/>)
+      }
+    }
+  }
+
+  if(toRender.length === 0){
+    toRender.push(<p>Something is wrong, please <Link to='/institutions'>Go Back</Link></p>)
```

Comment:
We should say "please go back to your institutions" (or something like that). "Back" might not be that page in all cases, so we should be clear about where we're sending them.